src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -473,6 +473,39 @@ uct_ib_mlx5_set_data_seg(struct mlx5_wqe_data_seg *dptr,
     dptr->addr       = htonll((uintptr_t)address);
 }
 
+static UCS_F_ALWAYS_INLINE
+unsigned uct_ib_mlx5_set_data_seg_iov(uct_ib_mlx5_txwq_t *txwq,
+                                      struct mlx5_wqe_data_seg *dptr,
+                                      const uct_iov_t *iov, size_t iovcnt)
+{
+    size_t   iov_it;
+    char    *dptr_it;
+    unsigned len = 0;
+
+    dptr_it = (char *)dptr;
+    for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
+        if (!iov[iov_it].length) { /* Skip zero length WQE*/
+            continue;
+        }
+        ucs_assert(iov[iov_it].memh != UCT_INVALID_MEM_HANDLE);
+
+        /* check consistency of the circular buffer */
+        if (ucs_unlikely((void*)dptr_it >= txwq->qend)) {
+            dptr_it = (void*)dptr_it - (txwq->qend - txwq->qstart);
```

Comment:
in this case, it's enough to do "if (dptr_it == txwq->qend) { dptr_it=txwq->qstart} " because dptr_it is always a 16-byte aligned pointer which cannot "miss" txwq->qend